edge-hub/watchdog/linux/src/child_process.rs

```diff
@@ -0,0 +1,62 @@
+use nix::{
+    sys::signal::{self, Signal},
+    unistd::Pid,
+};
+
+use std::{process::Child, thread::sleep, time::Duration};
+use tracing::{error, info};
+
+const PROCESS_SHUTDOWN_TOLERANCE_SECS: u64 = 5;
+pub const PROCESS_POLL_INTERVAL_SECS: u64 = 1;
+
+pub struct ChildProcess {
+    pub name: String,
```

Comment:
lets make a constructor function `pub fn run(name: String, process: Child) -> Self` instead of public members. it is usually a bad practice to the coupling on internal implementations